### PR TITLE
refactor: ConnectWallet v5 in dashboard 

### PR DESCRIPTION
**Pull request by jnsdls**
### TL;DR

Rewrite `useAddress` function with `useActiveAccount` function across various components to streamline account state handling. Replace `useChainId` with `useActiveWalletChain` to manage chain IDs more accurately. Update various other functions to align with the `thirdweb/react` package changes.

### What changed?

1. Replaced `useAddress` with `useActiveAccount` in multiple components.
2. Replaced `useChainId` with `useActiveWalletChain` in multiple components.
3. Updated hook imports and usages to align with the `thirdweb/react` package updates.
4. Revisited and updated certain functionalities, including wallet connections, contract interactions, and UI updates for better efficiency.

### How to test?

1. Check if the `useActiveAccount` function accurately reflects the current active account state across different components.
2. Verify the `useActiveWalletChain` function is managing and reflecting the chain ID appropriately.
3. Test various functionalities including wallet connections and contract interactions to ensure they are working as expected after the hook changes.

### Why make this change?

To streamline account and chain state handling, leveraging the latest hooks from the `thirdweb/react` package for better modernity and efficiency in code maintenance and functionality.

---

---

## PR-Codex overview
This PR focuses on improving loading performance for chains and enhancing user authentication and account management functionalities.

### Detailed summary
- Improved loading performance for chains
- Enhanced user authentication with `useActiveAccount`
- Replaced `useAddress` with `useActiveAccount` for better account management

> The following files were skipped due to too many changes: `apps/dashboard/src/pages/dashboard/settings/billing.tsx`, `apps/dashboard/src/contract-ui/tabs/tokens/components/burn-button.tsx`, `apps/dashboard/src/contract-ui/tabs/proposals/page.tsx`, `apps/dashboard/src/core-ui/nft-drawer/useNftDrawerTabs.tsx`, `apps/dashboard/src/contract-ui/tabs/tokens/components/transfer-button.tsx`, `apps/dashboard/src/contract-ui/tabs/tokens/components/burn-form.tsx`, `apps/dashboard/src/contract-ui/tabs/claim-conditions/components/claim-conditions-form/index.tsx`, `apps/dashboard/src/contract-ui/tabs/settings/components/royalties.tsx`, `packages/thirdweb/src/react/web/ui/components/ChainName.tsx`, `apps/dashboard/src/contract-ui/components/solidity-inputs/address-input.tsx`, `apps/dashboard/src/components/engine/contract-subscription/contract-subscriptions-table.tsx`, `apps/dashboard/src/contract-ui/tabs/listings/components/listing-drawer.tsx`, `apps/dashboard/src/contract-ui/tabs/tokens/components/airdrop-button.tsx`, `apps/dashboard/src/pages/api/moralis/balances.ts`, `apps/dashboard/src/contract-ui/tabs/tokens/components/mint-form.tsx`, `apps/dashboard/src/contract-ui/tabs/tokens/components/airdrop-form.tsx`, `apps/dashboard/src/components/contract-components/contract-deploy-form/index.tsx`, `apps/dashboard/src/contract-ui/tabs/shared-components/listing-drawer.tsx`, `apps/dashboard/src/contract-ui/tabs/settings/components/primary-sale.tsx`, `apps/dashboard/src/contract-ui/tabs/account-permissions/components/account-signer.tsx`, `apps/dashboard/src/pages/profile/[profileAddress].tsx`, `apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx`, `apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/layout.tsx`, `apps/dashboard/src/components/onboarding/Steps.tsx`, `apps/dashboard/src/contract-ui/tabs/shared-components/list-button.tsx`, `apps/dashboard/src/contract-ui/components/primary-dashboard-button.tsx`, `packages/thirdweb/src/react/native/ui/components/ChainIcon.tsx`, `apps/dashboard/src/@3rdweb-sdk/react/hooks/useEngine.ts`, `apps/dashboard/src/contract-ui/tabs/nfts/components/mint-supply-tab.tsx`, `apps/dashboard/src/contract-ui/tabs/split/page.tsx`, `apps/dashboard/src/contract-ui/tabs/nfts/components/airdrop-tab.tsx`, `packages/thirdweb/src/react/web/ui/components/TokenIcon.tsx`, `apps/dashboard/src/contract-ui/tabs/nfts/components/claim-tab.tsx`, `packages/thirdweb/src/react/web/ui/components/token/TokenSymbol.tsx`, `apps/dashboard/src/pages/dashboard/contracts/deploy.tsx`, `apps/dashboard/src/components/contract-components/published-contract/index.tsx`, `apps/dashboard/src/contract-ui/tabs/english-auctions/components/table.tsx`, `apps/dashboard/src/pages/dashboard/contracts/publish.tsx`, `apps/dashboard/src/components/contract-components/contract-publish-form/index.tsx`, `apps/dashboard/src/app/(dashboard)/providers.tsx`, `apps/dashboard/src/components/engine/engine-instances-table.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/swap/BuyTokenInput.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/tx-history/TokenInfoRow.tsx`, `apps/dashboard/src/contract-ui/tabs/direct-listings/components/table.tsx`, `apps/dashboard/src/components/contract-tabs/code/ContractCode.tsx`, `apps/dashboard/src/components/cmd-k-search/index.tsx`, `apps/dashboard/src/contract-ui/tabs/nfts/components/mint-form.tsx`, `packages/thirdweb/src/react/native/ui/connect/SendScreen.tsx`, `apps/dashboard/src/contract-ui/tabs/settings/components/metadata.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/swap/ConfirmationScreen.tsx`, `apps/dashboard/src/contract-ui/tabs/code/components/code-overview.tsx`, `apps/dashboard/src/@3rdweb-sdk/react/hooks/useRegistry.ts`, `apps/dashboard/src/contract-ui/tabs/settings/components/platform-fees.tsx`, `packages/thirdweb/src/chains/utils.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/tx-history/BuyTxHistory.tsx`, `packages/thirdweb/src/react/native/ui/connect/ConnectedModal.tsx`, `apps/dashboard/src/pages/dashboard/index.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/swap/PayWithCrypto.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/fiat/FiatTxDetailsTable.tsx`, `apps/dashboard/src/contract-ui/tabs/overview/components/MarketplaceDetails.tsx`, `apps/dashboard/src/components/engine/overview/backend-wallets-table.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/TokenSelector.tsx`, `apps/dashboard/src/@3rdweb-sdk/react/hooks/useVote.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/BuyScreen.tsx`, `apps/dashboard/src/components/ipfs-upload/dropzone.tsx`, `apps/dashboard/src/components/app-layouts/app.tsx`, `apps/dashboard/src/components/contract-components/import-contract/modal.tsx`, `apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/components/client/add-chain-to-wallet.tsx`, `apps/dashboard/src/components/wallets/PosthogIdentifier.tsx`, `apps/dashboard/src/@3rdweb-sdk/react/hooks/useContractRoles.ts`, `apps/dashboard/src/components/onboarding/index.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Details.tsx`, `apps/dashboard/src/@3rdweb-sdk/react/hooks/useLoggedInUser.ts`, `apps/dashboard/src/components/selects/CustomChainRenderer.tsx`, `apps/dashboard/src/pages/cli/login.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/tx-history/SwapDetailsScreen.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/fiat/FiatSteps.tsx`, `apps/dashboard/src/components/app-layouts/providers.tsx`, `apps/dashboard/src/components/buttons/TransactionButton.tsx`, `apps/dashboard/src/components/contract-components/hooks.ts`, `packages/thirdweb/src/react/core/hooks/others/useChainQuery.ts`, `apps/dashboard/src/components/buttons/MismatchButton.tsx`, `apps/dashboard/src/components/selects/NetworkSelectorButton.tsx`, `apps/dashboard/src/components/app-layouts/provider-setup.tsx`, `apps/dashboard/src/@3rdweb-sdk/react/components/connect-wallet/index.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by jnsdls** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3397?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3397** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3397?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3392** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3392?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 3 other dependent PRs ([#3393](https://github.com/thirdweb-dev/js/pull/3393) <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3393?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#3395](https://github.com/thirdweb-dev/js/pull/3395) <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3395?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#3396](https://github.com/thirdweb-dev/js/pull/3396) <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3396?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jnsdls and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by jnsdls** [comment]
we're waiting for #3537 to ship first tomorrow

File: apps/dashboard/src/@3rdweb-sdk/react/components/connect-wallet/index.tsx [hunk 1]
```diff
@@ -1,85 +1,182 @@
...
+      auth={authConfig}
       theme={t}
-      welcomeScreen={() => {
-        return <ConnectWalletWelcomeScreen theme={t} />;
+      client={thirdwebClient}
+      connectModal={{
+        privacyPolicyUrl: "/privacy",
+        termsOfServiceUrl: "/tos",
+        showThirdwebBranding: false,
+        welcomeScreen: () => <ConnectWalletWelcomeScreen theme={t} />,
```
Comment (joaquim-verges) [inline apps/dashboard/src/@3rdweb-sdk/react/components/connect-wallet/index.tsx:162]: btw not right now, but i think short term we might want to get rid of the welcome screen to just promote the inApp wallet login more

File: apps/dashboard/src/components/app-layouts/provider-setup.tsx [hunk 2]
```diff
@@ -103,52 +64,60 @@ export const DashboardThirdwebProviderSetup: ComponentWithChildren<
...
+      storageInterface={StorageSingleton}
+    >
+      {children}
+    </ThirdwebSDKProvider>
   );
 };
 
-const GlobalAuthTokenProvider = () => {
-  const { token, isLoading } = useApiAuthToken();
+function useEthersSigner() {
```
Comment (joaquim-verges) [inline apps/dashboard/src/components/app-layouts/provider-setup.tsx:88]: wonder if we should just have this as a 'adapter hook' exported out from the sdk

File: packages/thirdweb/src/react/core/hooks/others/useChainQuery.ts [hunk 3]
```diff
@@ -1,49 +1,130 @@
...
-    });
-  }, [chains, maxConcurrency]);
-
-  return useQueries({
-    queries: queryList,
+export function useChainExplorers(chain?: Chain) {
+  // only if we have a chain and it might be a testnet and no faucets and its not localhost
+  const isEnabled = !!chain && !chain.blockExplorers?.length;
+
+  const chainQuery = useQuery({
```
Comment (joaquim-verges) [inline packages/thirdweb/src/react/core/hooks/others/useChainQuery.ts:112]: this useQuery could be extracted out into 1 function that takes in  `isEnabled`

File: packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx [hunk 4]
```diff
@@ -259,57 +157,182 @@ function NetworkSelectorContentInner(
...
+    othersLabel,
+  ]);
 
-  const recentChainsFiltered = useMemo(() => {
-    if (!fuseInstances) {
-      return recentChainsTab;
-    }
+  // fuse instance for searching
+  const fuse = useMemo(() => {
+    return new Fuse(allChains, {
```
Comment (joaquim-verges) [inline packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx:270]: this is react native friendly right?

File: packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx [hunk 5]
```diff
@@ -830,3 +757,174 @@ const StyledMagnifyingGlassIcon = /* @__PURE__ */ styled(MagnifyingGlassIcon)(
...
+ *          { label: 'Popular', chains: [base, ethereum, sepolia] },
+ *        ]
+ *     });
+ *   }
+ *
+ *   return <button onClick={handleClick}> Switch Network </button>
+ * }
+ * ```
+ */
+export function useNetworkSwitcherModal() {
```
Comment (joaquim-verges) [inline packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx:879]: noice

**Review by joaquim-verges — APPROVED** [review]

File: packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx (hunk 4 above)
Comment (jnsdls) [reply packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx:270]: yeah fuse is pure js

File: apps/dashboard/src/components/app-layouts/provider-setup.tsx (hunk 2 above)
Comment (jnsdls) [reply apps/dashboard/src/components/app-layouts/provider-setup.tsx:88]: eh... maybe? Can add it as a follow-up

**Pull request merged**